torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -374,14 +380,18 @@ struct NCCLTraceBuffer {
         torch::CapturedTraceback::gather(true, true, capture_cpp_stack_);
     std::lock_guard<std::mutex> guard(mutex_);
 
+    static c10::ApproximateClockToUnixTimeConverter clock_converter;
+    static auto tsc_to_ns = clock_converter.makeConverter();
+
     auto te = Entry{
         id_,
         pg_id,
         seq_id,
         profiling_name,
         std::move(traceback),
         std::move(start),
-        std::move(end)};
+        std::move(end),
+        uint64_t(tsc_to_ns(c10::getApproximateTime()) / 1000)};
```

Comment:
and we store just getApproximateTime here.

Reply:
yep, agreed.